src/main/java/uk/gov/pay/directdebit/payments/params/PaymentViewSearchParams.java

```diff
@@ -171,29 +172,29 @@ public String generateQuery() {
     }
     
     public String buildQueryParamString() {
-        StringBuilder sb = new StringBuilder("");
+        String query = "";
         if (isNotBlank(mandateId)) {
-            sb.append("&" + MANDATE_ID_KEY + "=" + mandateId);
+            query += "&" + MANDATE_ID_KEY + "=" + mandateId;
         }
         if (searchDateParams != null) {
             if (searchDateParams.getFromDate() != null) {
-                sb.append("&" + FROM_DATE_KEY + "=" + searchDateParams.getFromDate().toString());
+                query += "&" + FROM_DATE_KEY + "=" + searchDateParams.getFromDate().toString();
             }
             if (searchDateParams.getToDate() != null) {
-                sb.append("&" + TO_DATE_KEY + "=" + searchDateParams.getToDate().toString());
+                query += "&" + TO_DATE_KEY + "=" + searchDateParams.getToDate().toString();
             }
         }
         if (isNotBlank(email)) {
-            sb.append("&" + EMAIL_FIELD + "=" + email);
+            query += "&" + EMAIL_FIELD + "=" + email;
         }
         if (isNotBlank(reference)) {
-            sb.append("&" + REFERENCE_FIELD + "=" + reference);
+            query += "&" + REFERENCE_FIELD + "=" + reference;
         }
         if (amount != null) {
-            sb.append("&" + AMOUNT_FIELD + "=" + amount);
+            query += "&" + AMOUNT_FIELD + "=" + amount;
         }
-        sb.append(addPaginationParams());
-        return sb.toString();
+        query += addPaginationParams();
```

Comment:
If I’m remembering my Java internals correctly, this will cause lots of `StringBuilder`s to be created: the compiler isn’t smart enough to use the same `StringBuilder` across multiple statements.

Where we have lots of conditionals mixed in, it’s best to create a `StringBuilder` and then always use `append` to add to it (e.g. `sb.append("&").append(MANDATE_ID_KEY).append("=").append(mandateId)` etc.).

Not that any of this is likely to make a huge amount of difference to performance!